src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfEhFrame.cs

```diff
@@ -0,0 +1,203 @@
+                augmentationLength++;
+            }
+            if (cie.IsSignalFrame)
+            {
+                augmentationString.Append('S');
+            }
+
+            uint length =
+                4u + // Length
+                4u + // CIE Offset (0)
+                1u + // Version
+                (uint)augmentationString.Length + 1u +
+                DwarfHelper.SizeOfULEB128(cie.CodeAlignFactor) +
+                DwarfHelper.SizeOfSLEB128(cie.DataAlignFactor) +
+                DwarfHelper.SizeOfULEB128(cie.ReturnAddressRegister) +
+                (uint)(augmentationLength > 0 ? DwarfHelper.SizeOfULEB128(augmentationLength) + augmentationLength : 0) +
+                (uint)cie.Instructions.Length;
+            uint padding = ((length + 7u) & ~7u) - length;
+
+            BinaryPrimitives.WriteUInt32LittleEndian(tempBuffer, length + padding - 4u);
+            _sectionWriter.Write(tempBuffer);
+
+            _sectionWriter.WriteByte(cie.ReturnAddressRegister < 0x7F ? (byte)1u : (byte)3u); // Version
+            _sectionWriter.Write(augmentationString.UnderlyingArray);
+
+            _sectionWriter.WriteULEB128(cie.CodeAlignFactor);
+            _sectionWriter.WriteSLEB128(cie.DataAlignFactor);
+            _sectionWriter.WriteULEB128(cie.ReturnAddressRegister);
+
+            _sectionWriter.WriteULEB128(augmentationLength);
+            if (cie.PersonalitySymbolName != null)
+            {
+                _sectionWriter.WriteByte(cie.PersonalityEncoding);
+                WriteAddress(cie.PersonalityEncoding, cie.PersonalitySymbolName);
+            }
+            if (cie.LsdaEncoding != 0)
+            {
+                _sectionWriter.WriteByte(cie.LsdaEncoding);
+            }
+            if (cie.PointerEncoding != 0)
+            {
+                _sectionWriter.WriteByte(cie.PointerEncoding);
+            }
+
+            _sectionWriter.Write(cie.Instructions);
+
+            _sectionWriter.Write(stackalloc byte[(int)padding]);
+        }
+
+        private void WriteFde(DwarfFde fde, uint cieOffset)
+        {
+            Span<byte> tempBuffer = stackalloc byte[8];
+
+            uint augmentationLength =
+                fde.Cie.FdesHaveAugmentationData ?
+                    1u + // Length
+                    (fde.Cie.PersonalityEncoding != 0 ? AddressSize(fde.Cie.PersonalityEncoding) : 0) +
+                    (fde.Cie.LsdaEncoding != 0 ? AddressSize(fde.Cie.LsdaEncoding) : 0) : 0;
+
+            uint length =
+                4u + // Length
+                4u + // CIE offset
+                AddressSize(fde.Cie.PointerEncoding) + // PC start
+                AddressSize(fde.Cie.PointerEncoding) + // PC end
+                augmentationLength +
+                (uint)fde.Instructions.Length;
+            uint padding = ((length + 7u) & ~7u) - length;
+
+            BinaryPrimitives.WriteUInt32LittleEndian(tempBuffer, length + padding - 4u);
+            _sectionWriter.Write(tempBuffer.Slice(0, 4));
```

Comment:
Are we just writing 32bit LE values or there is more going on here?

Reply:
Right. There's a helper method for that now, I should use it 😅

_The code predates the existence of the helper method and I missed this place during the cleanup._